### ISSUE TITLE
problems about axis of concat layer when convert pytorch to caffe

### ISSUE THREAD

**Issue by hchnu** [microsoft/MMdnn#769]
Platform (like ubuntu 16.04/win10):
ubuntu16.04
Python version:
3.5.5
Source framework with version (like Tensorflow 1.4.1 with GPU):
pytorch
Destination framework with version (like CNTK 2.3 with GPU):
caffe
Pre-trained model path (webpath or webdisk path):

Running scripts:

Hello, everyone.
I have a question about the **Concat** layer when i convert **pytorch** models to **caffe** models. In the pytorch_parser.py:

```
def rename_Concat(self, source_node):
        IR_node = self._convert_identity_operation(source_node, new_op='Concat')
        if source_node.attrs['axis'] == 1:
            IR_node.attr['axis'].i = len(self.shape_dict[source_node.name]) - 1
        else:
            IR_node.attr['axis'].i = source_node.attrs['axis']

```
 And in the caffe_emitter.py:

```
def emit_Concat(self, IR_node):
        axis_array = (2, 3, 1, 0)
        axis = axis_array.index(IR_node.get_attr('axis'))
        input_layers = ', '.join(('n.' + self.IR_graph.get_node(edge).real_variable_name) for edge in IR_node.in_edges)
        self.add_body(1, "n.{:<15} = L.Concat({}, axis={})".format(
            IR_node.variable_name,
            input_layers,
            axis
        ))

```
So what does 'axis' really mean? Why do you modify the value of 'axis'?

**Comment by rainLiuplus** [comment]
Hi @hchnu,
It is because that the layout of TensorFlow can be channel last (NHWC), but other frameworks only support NCHW. We let the IR be a certain format, NHWC; So, if the target format is NCHW, we transpose the input data and modify the related operators to satisfy the required layout.

**Comment by hchnu** [comment]
@rainLiuplus 
Thank you for your answer very much
